### ISSUE TITLE
HOI3 Compatibility? Request

### ISSUE THREAD

**Issue by reidellawfirm** [Zeon8/HOI4Launcher#1]
Is this compatible with launching HOI3 and mods? I tried to change directory to HOI3 folder and the launcher crashes when I try to open HOI3.

If it's possible to make this compatible with HOI3 or fork a version for it, I would like to sponsor it. Let me know if we can make this happen.

**Comment by Zeon8** [comment]
No, it is not compatible with HOI3 and to make it possible it needs full rewrite. Besides, HOI3 mods don't have icons to display them, and the official game launcher starts fast enough and works just fine. So there is no benefit in creating another launcher for HOI3.

**Comment by reidellawfirm** [comment]
Ah, gotcha. So I have a unique use case for a HOI3 launcher. If you would be interested I would pay to have it developed to test for my use case.

For my use case, I am running HOI3 on Mobox on android and the Paradox Launcher fails to open within Mobox. I haven't quite nailed down the issue with why it won't open but the game runs fine being opened directly but I have no method of running mods when the game starts directly. 

Your HOI4 Launcher opens great within Mobox but is not compatible with HOI3, so I am unable to use it to load mods and run HOI3. 

So the only solution I am seeking is to load mods and run HOI3. I think potentially your Launcher for HOI3 would do that. 

The Irony Mod Manager also fails to open like the Paradox Launcher within Mobox.

**Comment by Zeon8** [comment]
Sorry that it took me so long to answer. I have created [new launcher for HOI3](https://github.com/Zeon8/Hoi3Launcher). I haven't tried to run it on Mobox, but it works on Winlator. 
Before starting the launcher, you should set `DOTNET_GCHeapHardLimit=0x1C00000` environment variable.
If you would like, you can [sponsor me](https://ko-fi.com/zeon8) for creating this launcher.

**Comment by reidellawfirm** [comment]
Amazing, thank you for that. I will be able to test it this weekend and see if that works.

I did try your sponsor link but PayPal wasn't processing either credit card nor my account for some reason. I'll try again later today from the office but worst case you can send me your PayPal email and I'll just directly send it to you.

**Comment by reidellawfirm** [comment]
I have tried every which way to donate through your link but I keep getting errors that state Paypal isn't working. Do you have a paypal email or user I can just send it direct to you?

I haven't been able to get the launcher to work yet in Winlator or Mobox but I am playing around with various settings. What were your settings in winlator?

**Comment by Zeon8** [comment]
I have created a new [donation page for PayPal](https://www.paypal.com/donate/?hosted_button_id=FMHB6K59UP3FN).

~~Make sure you set `DOTNET_GCHeapHardLimit` variable to `1C0000000`.~~
I have just updated the launcher so that it is no longer requires setting `DOTNET_GCHeapHardLimit` variable. Please redownload files from release page and extract all files to HOI3 folder like this:
![image](https://github.com/user-attachments/assets/cbbdf2bc-da82-4311-97de-a2c3499321d8)

Launcher should work for both Turnip and VirGL driver.

<details>
<summary>My settings:</summary>

![zb2rhXa78Bs5KGhjU9WePp5uGvB5uhSdJu4YQZyaw9k3kMgqA_eeed89b3d047cec0e4a91fdae3a38d355c8fc4b7](https://github.com/user-attachments/assets/9401540e-829e-471a-aea9-d7ba715320cb)
</details>

**Comment by reidellawfirm** [comment]
Thank you, will test again shortly. I did have your launcher in a different folder and that was probably the issue. 

I still had trouble with your payment link, this time it gave me an error stating transfer was not allowed due to regulations. But either way, I think I found you on PayPal and was able to send it as a friend, hope you got it.

**Comment by Zeon8** [comment]
Thank you so much for your generous donation! I am really glad I was able to help and I truly appreciate your support.

**Comment by reidellawfirm** [comment]
No worries, appreciate your time and efforts. I am still testing the HOI3 launcher in mobox. I can get it to open but it doesn't actually start the mod but the base game. After I run a few more tests, I will go ahead and ping you on the other launcher github with any potential requests. Though I think it is probably the mod and not the launcher.
